cpp/src/table/legacy/table.cpp

```diff
@@ -97,6 +97,24 @@ void table::destroy(void) {
   }
 }
 
+table table::select(std::vector<gdf_size_type> const& column_indices){
+    CUDF_EXPECTS(column_indices.size() <= num_columns(), "Requested too many columns.");
+    std::vector<gdf_column*> desired_columns;
+    for(auto index : column_indices){
+        desired_columns.push_back(get_column(index));
+    }
+    return table{desired_columns};
+}
+
+table table::select(std::vector<gdf_size_type> const& column_indices) const {
+    CUDF_EXPECTS(column_indices.size() <= num_columns(), "Requested too many columns.");
+    std::vector<gdf_column*> desired_columns;
+    for(auto index : column_indices){
+        desired_columns.push_back(const_cast<gdf_column*>(get_column(index)));
```

Comment:
I realize if we just access `_columns` directly, then there's no need for a `const_cast`. Furthermore, there's no need for a `const` and `non-const` version of `select`, we can just have the single `const` version.